source/extensions/access_loggers/common/grpc_access_logger.h

```diff
@@ -127,6 +136,9 @@ template <typename LogRequest, typename LogResponse> class GrpcAccessLogClient {
       } else {
         stream_->stream_->sendMessage(request, false);
       }
+      if (isStreamStarted() && !streaming_) {
+        stream_->stream_->closeStream();
```

Comment:
I think this may not always work, as it relies on the access log server to close the stream before the new stream can be created. This can race. For example if the server is too slow there maybe two or more requests sent on the stream before remote close is registered by Envoy. I think I will need a second pair of eyes to look into this change. I will need to find someone else to look at this.